scripts/calculateRevenue/protocols/beefy/index.ts

```diff
@@ -45,37 +49,77 @@ export function getVaultTvl(
   return lastTvl
 }
 
+/**
+ * Given a list of historical token prices and a timestamp, returns the nearest token price
+ * to the specified time.
+ * Requires the list of tokenPrices to be sorted from earliest to most-recent.
+ */
+export function getTokenPrice(
+  tokenPrices: TokenPriceData[],
+  timestamp: Date,
+): number {
+  let lastPrice = tokenPrices[0].priceUsd
+  for (const tokenPrice of tokenPrices) {
+    if (new Date(tokenPrice.priceFetchedAt) >= timestamp) {
+      return Number(lastPrice)
+    }
+    lastPrice = tokenPrice.priceUsd
+  }
+  return Number(lastPrice)
+}
 /**
  * Given historical information about a vault and a user's transactions, calculates the amount of fee
  * revenue generated by the user's interaction with the vault, denominated in the native currency
  * of the chain the vault is deployed on.
  */
-export async function calculateVaultRevenue(vaultInfo: VaultInfo): Promise<{
-  tokenId: string
-  revenue: string
-}> {
-  let totalRevenueContribution = 0n
-  for (const { beefyFee, timestamp } of vaultInfo.feeEvents) {
-    const userTvl = getUserTvl(vaultInfo.txHistory, timestamp)
-    const vaultTvl = getVaultTvl(vaultInfo.vaultTvlHistory, timestamp)
-    const partialRevenueContribution =
-      (BigInt(userTvl * 10 ** 18) * BigInt(beefyFee)) /
-      BigInt(vaultTvl * 10 ** 18)
-    // TODO: Convert fee contributions to USD value using native token price at the time of fee collection
-    // using an historical API endpoint, e.g. https://docs.coingecko.com/reference/coins-id-history
-    totalRevenueContribution += partialRevenueContribution
-  }
-
+export async function calculateVaultRevenue(
+  vaultInfo: VaultInfo,
+): Promise<number> {
   const strategyContract = await getStrategyContract(
     vaultInfo.vaultAddress,
     vaultInfo.networkId,
   )
   const nativeTokenAddress = await strategyContract.read.native()
-  const tokenId = `${vaultInfo.networkId}:${nativeTokenAddress}`
-  return {
+  const tokenId = `${vaultInfo.networkId}:${nativeTokenAddress.toLowerCase()}`
```

Comment:
aren't native tokens `<network-id>:native`?

Reply:
Ah good call. For chains where the native token is something like wrapped ETH, we can use the address, but `native` is more universal.